File: frontend/views/vista-act01usuarionoregistrado.ts
import { LitElement, html, css, customElement } from 'lit-element';
import '@vaadin/vaadin-ordered-layout/src/vaadin-horizontal-layout.js';
import '@vaadin/vertical-layout/src/vaadin-vertical-layout.js';
import '@polymer/iron-icon/iron-icon.js';
import '@vaadin/button/src/vaadin-button.js';

@customElement('vista-act01usuarionoregistrado')
export class VistaAct01usuarionoregistrado extends LitElement {
  static get styles() {
    return css`
      :host {
          display: block;
          height: 100%;
      }
      `;
  }

  render() {
    return html`
<vaadin-horizontal-layout class="content; " style="width: 100%; height: 100%; position:absolute" id="horizontalLayoutPrincipalNoRegistrado">
 <vaadin-vertical-layout theme="spacing" style="flex-grow: 1;" id="verticalLayoutIzquierdoNoRegistrado">
  <span style="align-self: center; padding-top: var(--lumo-space-s);" id="spanListaUsuarios">Lista de usuarios</span>
  <vaadin-vertical-layout theme="spacing" style="flex-grow: 1; align-self: center; width: 100%;" id="verticalLayoutListaUsuarios"></vaadin-vertical-layout>
  <vaadin-button style="align-self: center; margin-top: var(--lumo-space-m);" tabindex="0" id="botonMasUsuarios">
    Más usuarios...
  </vaadin-button>
 </vaadin-vertical-layout>
 <vaadin-vertical-layout theme="spacing" style="flex-grow: 1;" id="verticalLayoutCentralNoRegistrado">
  <vaadin-vertical-layout theme="spacing" id="verticalLayoutImagenNoRegistrado" style="align-self: center;">
   <img style="width: 25%; align-self: center; margin-top: var(--lumo-space-xl);" src="./img/imagenIcon.png">
  </vaadin-vertical-layout>
  <vaadin-button style="align-self: center;" tabindex="0" id="botonIniciarSesion">
    Iniciar Sesión 
  </vaadin-button>
  <vaadin-button style="align-self: center;" tabindex="0" id="botonRegistrarse">
    Registrarse 
  </vaadin-button>
  <vaadin-button style="align-self: center;" tabindex="0" id="botonIniciarSesionGoogle">
   <iron-icon icon="lumo:arrow-right" slot="suffix" id="ironIcon"></iron-icon>Iniciar sesión con <iron-icon src="./img/googleIcon.png"></iron-icon>
  </vaadin-button>
 </vaadin-vertical-layout>
 <vaadin-vertical-layout theme="spacing" style="flex-grow: 1;" id="verticalLayoutDerechoNoRegistrado">
  <span style="align-self: center; padding-top: var(--lumo-space-s);" id="spanListaHashtagsNoRegistrado">Lista de hashtags</span>
  <vaadin-vertical-layout theme="spacing" style="flex-grow: 1; align-self: center; width: 100%;" id="verticalLayoutListaHashtagsNoRegistrado"></vaadin-vertical-layout>
  <vaadin-button style="align-self: center; margin-top: var(--lumo-space-m);" tabindex="0" id="botonMasHashtags">
    Más hashtags...
  </vaadin-button>
 </vaadin-vertical-layout>
</vaadin-horizontal-layout>
`;
  }

  // Remove this method to render the contents of this view inside Shadow DOM
  createRenderRoot() {
    return this;
  }
}

File: twitter-vaadin-grupo-20/src/main/java/vistas/VistaAct01usuarionoregistrado.java
package vistas;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.dom.Element;
import com.vaadin.flow.component.button.Button;

@Tag("vista-act01usuarionoregistrado")
@JsModule("./views/vista-act01usuarionoregistrado.ts")
public class VistaAct01usuarionoregistrado extends LitTemplate {

    @Id("verticalLayoutListaUsuarios")
    private Element verticalLayoutListaUsuarios;
    @Id("verticalLayoutCentralNoRegistrado")
    private Element verticalLayoutCentralNoRegistrado;
    @Id("verticalLayoutListaHashtagsNoRegistrado")
    private Element verticalLayoutListaHashtagsNoRegistrado;
    @Id("botonIniciarSesion")
    private Button botonIniciarSesion;
    @Id("botonRegistrarse")
    private Button botonRegistrarse;
    @Id("botonMasUsuarios")
    private Button botonMasUsuarios;
    @Id("botonMasHashtags")
    private Button botonMasHashtags;

    public VistaAct01usuarionoregistrado() {
        // You can initialise any data required for the connected UI components here.
    }

    public Element getVerticalLayoutListaUsuarios() {
        return verticalLayoutListaUsuarios;
    }

    public Element getVerticalLayoutCentralNoRegistrado() {
        return verticalLayoutCentralNoRegistrado;
    }

    public Element getVerticalLayoutListaHashtagsNoRegistrado() {
        return verticalLayoutListaHashtagsNoRegistrado;
    }

    public Button getBotonIniciarSesion() {
        return botonIniciarSesion;
    }

    public Button getBotonRegistrarse() {
        return botonRegistrarse;
    }

    public Button getBotonMasUsuarios() {
        return botonMasUsuarios;
    }

    public Button getBotonMasHashtags() {
        return botonMasHashtags;
    }

    // Remove this method to render the contents of this view inside Shadow DOM
    public com.vaadin.flow.component.Component createRenderRoot() {
        return this;
    }
}